libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -788,56 +570,89 @@ void AC_AutoTune::control_attitude()
+                    tune_yaw_rp = MAX(get_rp_min(), tune_yaw_rp * AUTOTUNE_RD_BACKOFF);
                     break;
                 }
                 break;
             case RP_UP:
-                tune_type = TuneType(tune_type + 1);
                 switch (axis) {
                 case ROLL:
-                    tune_roll_rp = MAX(AUTOTUNE_RP_MIN, tune_roll_rp * AUTOTUNE_RP_BACKOFF);
+                    tune_roll_rp = MAX(get_rp_min(), tune_roll_rp * AUTOTUNE_RP_BACKOFF);
                     break;
                 case PITCH:
-                    tune_pitch_rp = MAX(AUTOTUNE_RP_MIN, tune_pitch_rp * AUTOTUNE_RP_BACKOFF);
+                    tune_pitch_rp = MAX(get_rp_min(), tune_pitch_rp * AUTOTUNE_RP_BACKOFF);
                     break;
                 case YAW:
-                    tune_yaw_rp = MAX(AUTOTUNE_RP_MIN, tune_yaw_rp * AUTOTUNE_RP_BACKOFF);
+                    tune_yaw_rp = MAX(get_rp_min(), tune_yaw_rp * AUTOTUNE_RP_BACKOFF);
                     break;
                 }
                 break;
             case SP_DOWN:
-                tune_type = TuneType(tune_type + 1);
                 break;
             case SP_UP:
+                switch (axis) {
+                case ROLL:
+                    tune_roll_sp = MAX(get_sp_min(), tune_roll_sp * AUTOTUNE_SP_BACKOFF);
+                    // trad heli uses original parameter value rather than max demostrated through test
+                    if (set_accel_to_max_test_value()) {
+                        tune_roll_accel = MAX(AUTOTUNE_RP_ACCEL_MIN, test_accel_max * AUTOTUNE_ACCEL_RP_BACKOFF);
+                    }
+                    break;
+                case PITCH:
+                    tune_pitch_sp = MAX(get_sp_min(), tune_pitch_sp * AUTOTUNE_SP_BACKOFF);
+                    // trad heli uses original parameter value rather than max demostrated through test
+                    if (set_accel_to_max_test_value()) {
+                        tune_pitch_accel = MAX(AUTOTUNE_RP_ACCEL_MIN, test_accel_max * AUTOTUNE_ACCEL_RP_BACKOFF);
+                    }
+                    break;
+                case YAW:
+                    tune_yaw_sp = MAX(get_sp_min(), tune_yaw_sp * AUTOTUNE_SP_BACKOFF);
+                    // trad heli uses original parameter value rather than max demostrated through test
+                    if (set_accel_to_max_test_value()) {
+                        tune_yaw_accel = MAX(AUTOTUNE_Y_ACCEL_MIN, test_accel_max * AUTOTUNE_ACCEL_Y_BACKOFF);
+                    }
+                    break;
+                }
+                break;
+            case RP_DOWN:
+            case RFF_UP:
+            case RFF_DOWN:
+            case MAX_GAINS:
+            case TUNE_COMPLETE:
+                break;
+            }
+
+            // increment the tune type to the next one in tune sequence
+            tune_seq_curr++;
+            tune_type = tune_seq[tune_seq_curr];
```

Comment:
I think we might be better off with
```suggestion
            // move to next tune type in tune sequence
            tune_type = next_tune_type(tune_type);
```

where `next_tune_type` is

```
virtual TuneType next_tune_type(TuneType t) const = 0;
```

on `AC_AutoTune`

That method would be a simple `switch` on Multi (potentially just the dodgy `+1` crap we already use in `master`, however).  The Heli one looks more complicated based on parameters :-)

Reply:
There was a lot of thought that went behind this.  The idea was to allow the user to specify the sequence of tests they want to use whether it was just the FF tuning, Rate D and Rate P, Angle P or all of them.  However we decide to proceed, I want heli to be able to choose the sequence of tuning tests.

Reply:
Added a method as requested but wasn't able to do it as a function that returned a TuneType.  So just used the argument that returned the tune type.